modules/library/referencing/src/main/java/org/geotools/geometry/GeometryBuilder.java

```diff
@@ -273,22 +274,39 @@ public Curve createCurve(List segments) throws MismatchedReferenceSystemExceptio
 		// - Start and end params for the CurveSegments will be set
 		return getPrimitiveFactory().createCurve(segments);
 	}
-    public Curve createCurve(PointArray points) throws MismatchedReferenceSystemException, MismatchedDimensionException {
+	
+	@Deprecated
+	public Curve createCurve(PointArray points) throws MismatchedReferenceSystemException, MismatchedDimensionException {
+		return createCurve(points, true);
+    }
+	
+    public Curve createCurve(PointArray points, boolean closed) throws MismatchedReferenceSystemException, MismatchedDimensionException {
         if (points == null)
             throw new NullPointerException("Points are required to create a curve");
 
         // A curve will be created
         // - The curve will be set as parent curves for the Curve segments
         // - Start and end params for the CurveSegments will be set
         List/*<LineSegment>*/ segmentList = new ArrayList/*<LineSegment>*/();
-        for( int i=0; i<points.size();i++){
+        for( int i = 0; i < points.size() - 1; i++){
             int start = i;
-            int end = (i+1)%points.size();
+            int end = i + 1;
             DirectPosition point1 = points.getDirectPosition( start, null );
             DirectPosition point2 = points.getDirectPosition( end, null );
             LineSegment segment = createLineSegment( point1, point2 );
             segmentList.add( segment );
         }
+        
+        if(closed) {
+        	LineSegment first = (LineSegment) segmentList.get(0);
+        	LineSegment last = (LineSegment) segmentList.get(segmentList.size() - 1);
+        	
+        	if(!first.getStartPoint().equals(last.getEndPoint())) {
+        		LineSegment segment = createLineSegment(last.getEndPoint(), first.getStartPoint());
+        		segmentList.add( segment );
```

Comment:
I like to reformat code that I change. I use the official GeoTools Eclipse formatter configuration ```build/eclipse/formatter.xml```. I select the block and use Ctrl-Shift-F in Eclipse to limit reformatting to the changed code (or whole method if I change most of it). I do not reformat entire files as this makes the changes too hard to read.